src/commands/doctor.rs

```diff
@@ -0,0 +1,96 @@
+use std::{fmt::Debug, path::PathBuf};
+
+use anyhow::Result;
+use clap::Parser;
+use dialoguer::{console::Emoji, Select};
+use futures::FutureExt;
+use spin_doctor::Diagnosis;
+
+#[derive(Parser, Debug)]
+#[clap(hide = true, about = "Run the Spin Doctor")]
+pub struct DoctorCommand {
+    #[clap(short = 'f', long, default_value = "spin.toml")]
+    file: PathBuf,
+}
+
+impl DoctorCommand {
+    pub async fn run(self) -> Result<()> {
+        eprintln!("{icon}The Spin Doctor is in.", icon = Emoji("📟 ", ""));
+        eprintln!("{icon}Checking {:?}...", self.file, icon = Emoji("🩺 ", ""));
+
+        let count = spin_doctor::Checkup::new(self.file)
+            .for_each_diagnosis(move |diagnosis, patient| {
+                async move {
+                    show_diagnosis(&*diagnosis);
+
+                    if let Some(treatment) = diagnosis.treatment() {
+                        let desc = match treatment.description(patient).await {
```

Comment:
Under what circumstances would it be reasonable to error creating the _description_?  It seems like the previous line has already determined the treatment, so the error "couldn't prepare treatment" seems odd... I would kind of expect if we know the treatment then describing it should be infallible...?

Reply:
But wouldn't (shouldn't?) that fail at the `.treatment()` stage?  It's the fact that we _have_ a treatment but barf when trying to _describe_ it that I find puzzling.  Like I'd expect the treatment to be kind of determined at this point, and describing it to be entirely deterministic.

Probably I need to go back and read how treatments work again...

Reply:
You would have to build the `Treatment::description` all the way back at `diagnose`, _or_ keep a copy of enough patient information in the `Diagnosis` to build it during `treatment` (surely a health record privacy violation). Either way your description could be out of date by the time you're ready to apply the treatment because - since `diagnose` can return multiple issues - other treatments may have already run in the interim.

Reply:
Thinking "out loud" (mechanical keyboard), it might make sense to parse the `manifest_doc` lazily for each diagnosis anyway to avoid issues with the file being modified in the middle of an interactive checkup...

Reply:
> You would have to build the Treatment::description all the way back at diagnose

Could you expand on why?  Is the idea that `diagnosis.treatment()` is a simple getter (the `Diagnosis` includes the treatment)?  This seems to run counter to your observation that "other treatments may have already run in the interim."  I assumed `treatment()` calculated the treatment lazily, and so could bundle the description into the Treatment object.  But I have not yet got round to that re-read so...

Sorry, I realise I am kind of obsessing over one not very crucial line; but I think it is hinting at a deeper misunderstanding (on my part) of how this stuff hangs together.  Feel free to drop this conversation if it's not useful to you.

Reply:
I think the issue is that the `manfiest` checks effectively do a dry-run of the treatment and use the resulting diff as the result of this function, and `description` just isn't a good name for that. I'll take another crack at this, probably splitting this into two methods where one does the dry-run "diff" and the other can be used for the treatment choice menu as you suggested in the other comment.